internal/monitors/haproxy/metadata.yaml

```diff
@@ -0,0 +1,763 @@
+monitors:
+  - monitorType: haproxy
+    doc: |
+      This monitor scrapes [HAProxy](http://www.haproxy.org/) statistics from an HTTP endpoint or a UNIX socket.
+      Requires HAProxy 1.8+.
+
+      <!--- SETUP --->
+      ### HTTP Endpoint Config
+      HAProxy stats must be enabled as described [here](https://www.haproxy.com/blog/exploring-the-haproxy-stats-page/).
+      Define the HAProxy monitor in the agent configuration file and provide the csv export url where HAProxy
+      stats are served in csv format. This monitor supports basic HTTP authentication. Simply provide the
+      username and password if required as shown below.
+      ```
+      ...
+      monitors:
+        - type: haproxy
+          url: "http://localhost:8404/stats?stats;csv"
+          username: "my username"
+          password: "my password"
+      ...
+      ```
+
+      Note: The HAProxy process level stats given by the [show info](https://cbonte.github.io/haproxy-dconv/1.8/management.html#9.3-show%20info) command
+      are not available at the HTTP endpoint. Scrape stats from the unix socket instead to include the process level metrics.
+
+      ### Unix Socket Config
+      The location of the HAProxy socket file is defined in the HAProxy config file, as in the following example:
+      ```
+      global
+          daemon
+          stats socket /var/run/haproxy.sock
+          stats timeout 2m
+      ```
+      For the given location of the socket file configure the HAProxy monitor url as such:
+      ```
+      ...
+      monitors:
+        - type: haproxy
+          url: "unix:/var/run/haproxy.sock"
+      ...
+      ```
+    dimensions:
+      proxy_name:
+        description: HAProxy configured proxy name. Possible values for listeners, frontends, backends, and servers.
+      service_name:
+        description: HAProxy configured service name. Possible values are FRONTEND for frontend, BACKEND for backend, configured name for server/listener.
+      process_num:
+        description: A number assigned to an HAProxy process instance (0 for first instance, 1 for second, ...).
+    metrics:
+#      pxname:
```

Comment:
Are these valid or not?

Reply:
For the commented out metrics, decide if they should be in or out and delete them if out.

Reply:
ok